x/staking/teststaking/service.go

```diff
@@ -0,0 +1,134 @@
+package teststaking
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/crypto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// Service is a structure which wraps the staking handler
+// and provides methods useful in tests
+type Service struct {
+	t *testing.T
+	h sdk.Handler
+	k keeper.Keeper
+
+	Ctx        sdk.Context
+	Commission stakingtypes.CommissionRates
+	// Coin Denomination
+	Denom string
+}
+
+// NewService creates staking Handler wrapper for tests
+func NewService(t *testing.T, ctx sdk.Context, k keeper.Keeper) *Service {
```

Comment:
I would prefer to avoid the word "Service". We use protobuf services a lot (grpc queries, adr-031), so I suggest to use "service" iif we're talking about a protobuf service.

Some ideas: `Helper`, `Handler`, `Controller`

Reply:
There was a discussion about it: https://github.com/cosmos/cosmos-sdk/pull/7477#discussion_r501176184

This helpers are everywhere without much meaning / context
* I'm against calling packages with super generic names like `helper`, `testhelper`, `common` unless it's a package with a very generic functionality for the whole module. (I know that you are asking here about the structure name / function name).
* Initially I called it `Handler` because it was wrapping a handler, but since it doesn't accept messages, but rather creates them and calls Handler, I was thinking about other name. I want to `HandlerWrapper` for similar reasons to above. Also it doesn't implement the Handler interface.
* Service was the most natural for me. It's a testing service. 
* If you think it will conflict with the protobuf services, then ok, let's rename. `Controller` is probably the closes one from the list. Other ideas would `Exec`, `Executor`, ...

Reply:
Agreed ServiceHelper or HandlerHelper. Something for tests should indicate that clearly in the name.

Reply:
testing is already in the package name. `teststaking.Controller`? though `teststaking.Helper` sounds better even though I would prefer to avoid this ;)

Reply:
Renamed to `Helper`.